framework/src/modules/chain/dpos/dpos.js

```diff
@@ -1,11 +1,27 @@
-const Delegates = require('./delegates');
+const { Delegates, EVENT_ROUND_FINISHED } = require('./delegates');
 
 module.exports = class Dpos {
-	constructor({ storage, logger }) {
+	constructor({ storage, logger, slots }) {
+		this.finalizedBlockRound = 0;
 		this.delegate = new Delegates({ storage, logger });
+		this.slots = slots;
+		this.delegate.on(EVENT_ROUND_FINISHED, () => {
+			this.onRoundFinish();
+		});
 	}
 
 	async getRoundDelegates(round) {
 		return this.delegates.getRoundDelegates(round);
 	}
+
+	async onBlockFinalized({ height }) {
+		this.finalizedBlockRound = this.slots.calcRound(height);
+	}
+
+	async onRoundFinish() {
+		const delegateListOffsetForRound = 2;
```

Comment:
shouldn't `2` be a configurable parameter?

Reply:
yes, this parameter is used in multiple places. It should be defined in process_V2.js which is out of scope of this PR. So, i merely set a private variable for it. Maybe I can add a TODO comment for it.